testsuite/MDAnalysisTests/converters/test_openmm_parser.py

```diff
@@ -130,6 +159,89 @@ class TestOpenMMTopologyParser(OpenMMTopologyBase):
     expected_n_bonds = 1922
 
 
+class TestOpenMMTopologyParserWithNoElements(OpenMMTopologyBase):
+    ref_filename = app.PDBFile(PDB).topology
+    for a in ref_filename.atoms():
+        a.element = None
+
+    expected_n_atoms = 47681
+    expected_n_residues = 11302
+    expected_n_segments = 1
+    expected_n_bonds = 25533
+
+    expected_attrs = [
+        "ids",
+        "names",
+        "resids",
+        "resnames",
+        "masses",
+        "bonds",
+        "chainIDs",
+    ]
+
+    def test_no_elements_warn(self):
+        wmsg = ("Element information is missing for all the atoms. "
+                "Elements attribute will not be populated. "
+                "Atomtype attribute will be guessed using atom "
+                "name and mass will be guessed using atomtype."
+                "See MDAnalysis.topology.guessers.")
+
+        with pytest.warns(UserWarning, match=wmsg):
+            mda_top = self.parser(self.ref_filename).parse()
+
+
+class TestOpenMMTopologyParserWithPartialElements():
+    parser = mda.converters.OpenMMParser.OpenMMTopologyParser
+    ref_filename = app.PDBFile(PDB).topology
+    expected_n_atoms = 47681
+    expected_n_residues = 11302
+    expected_n_segments = 1
+    expected_n_bonds = 25533
+
+    def test_with_partial_elements(self):
+        wmsg = (f"Element information missing for some atoms. "
+                f"These have been given an empty element record "
+                f"with their atomtype set to 'X' "
+                f"and their mass set to 0.0. "
+                f"If needed they can be guessed using "
+                f"MDAnalysis.topology.guessers.")
+        with pytest.warns(UserWarning, match=wmsg):
+            mda_top = self.parser(self.ref_filename).parse()
+            assert mda_top.types.values[3344] == 'X'
+            assert mda_top.types.values[3388] == 'X'
+            assert mda_top.elements.values[3344] == ''
+            assert mda_top.elements.values[3388] == ''
```

Comment:
From these two tests I'm assuming that `PDB` has some elements, but not all. It would be better to swap these tests then -- have `test_with_partial_elements` be in the class that subclasses  `OpenMMTopologyBase` without editing the OpenMM `ref_filename`. Editing this class attribute for the entire class `TestOpenMMTopologyParserWithNoElements` of tests is not great practice, as it might affect the other tests or future tests that we add to the class as well, or to any in the string of base classes (`OpenMMTopologyBase` and `ParserBase`).

Then, have a separate function that edits the OpenMM topology to have no elements and test that the right error message is raised. Only create and edit the OpenMM system inside the function to keep changes contained.

Reply:
Umm just to be clear, I have to do the following:
1) Move `test_with_partial_elements` in a class that subclasses ` OpenMMTopologyBase` (with no modification to `ref_filename`.
3) Make a separate function to test with no elements(by creating and testing the topology in the function itself)
4) Get rid of one redundant class that remains after all this.

Does this sound correct?

Reply:
Yeah, that sounds about right -- the goal is to have tests that make sense. For example, it makes sense for the partial elements check to happen on a file that only has partial elements, during a whole suite of other tests first defined in ` OpenMMTopologyBase` or `ParserBase`. It makes a little less sense to edit the PDB file for the entire suite of tests just to accommoodate one test -- that can happen inside one function and be isolated, to avoid potential undesired side effects on the other tests in the class.